docs/internals/optimizer.rst

```diff
@@ -1121,6 +1122,51 @@ The step LiteralRematerialiser is not required for correctness. It helps deal wi
 ``function f(x) -> y { revert(y, y} }`` where the literal ``y`` will be replaced by its value ``0``,
 allowing us to rewrite the function.
 
+.. _unused-store-eliminator:
+
+UnusedStoreEliminator
+^^^^^^^^^^^^^^^^^^^^^
+
+Optimizer component that removes redundant ``sstore`` and memory store statements.
+In case of an ``sstore``, if all outgoing code paths revert (due to an explicit ``revert()``, ``invalid()``, or infinite recursion) or
+lead to another ``sstore`` for which the optimizer can tell that it will overwrite the first store, the statement will be removed.
+However, if a there is read operation between the initial ``sstore`` and the revert, or the overwriting ``sstore``, the statement
```

Comment:
```suggestion
However, if a there is a read operation between the initial ``sstore`` and the revert, or the overwriting ``sstore``, the statement
```

Reply:
Can I suggest a change to your change :D 
`However, if a there is a read...`

Reply:
Ah wait. Just saw that extra `a` xD